test/jasmine/tests/axes_test.js

```diff
@@ -4376,6 +4376,33 @@ describe('Test axes', function() {
             })
             .then(done, done.fail);
         });
+        it('should respect axis title placement on relayout', function(done) {
+            function getPos(gd, sel) {
+                return d3Select(gd).select(sel).node().getBoundingClientRect();
+            }
+
+            // Tick position is < title position since 0 is at the top of the graph,
+            // rather than at the bottom. We're checking that the ticks and title don't overlap
+            function assertLayout() {
+                var titleTop = getPos(gd, '.xtitle').top;
+                var tickBottom = getPos(gd, '.xtick').bottom;
+                expect(tickBottom).toBeLessThan(titleTop + 2); // allow two pixels tolerance
```

Comment:
Just to note: this specific test looks fine, but only because:
(1) all the tick labels are essentially the same length. `select(...).node()` gives you only the first element, but the first one is unlikely to conflict with the axis title. Only the labels that are directly above the title can push it down.
(2) the tick labels are dense enough that even when you zoom in the title never fits between them. If that's not the case, when the title lines up right it can actually squeeze up between those labels.